### PR TITLE
Cache import glob expansion via EvaluationContext

### PR DESCRIPTION
**Pull request by cdmihai**

### CODE REVIEW

**Comment by cdmihai** [comment]
@dotnet-bot test Windows_NT Build for CoreCLR please
https://ci2.dot.net/job/Microsoft_msbuild/job/master/job/innerloop_Windows_NT_CoreCLR_prtest/706/

**Comment by cdmihai** [comment]
@dotnet-bot test Windows_NT Build for CoreCLR please
https://ci2.dot.net/job/Microsoft_msbuild/job/master/job/innerloop_Windows_NT_CoreCLR_prtest/706/

File: src/Build.UnitTests/Definition/ProjectEvaluationContext_Tests.cs [hunk 1]
```diff
@@ -216,7 +216,7 @@ public static IEnumerable<object> ContextPinsGlobExpansionCacheData
 
         [Theory]
         [MemberData(nameof(ContextPinsGlobExpansionCacheData))]
-        public void ContextCachesGlopExpansions(EvaluationContext.SharingPolicy policy, string[][] expectedGlobExpansions)
+        public void ContextCachesItemElementGlopExpansions(EvaluationContext.SharingPolicy policy, string[][] expectedGlobExpansions)
```
Comment (rainersigwald) [inline src/Build.UnitTests/Definition/ProjectEvaluationContext_Tests.cs]: typo: glop

File: src/Build.UnitTests/Definition/ProjectEvaluationContext_Tests.cs [hunk 2]
```diff
@@ -243,6 +243,46 @@ public void ContextCachesGlopExpansions(EvaluationContext.SharingPolicy policy,
...
+            </Project>",
+
+            @"<Project>
+                <Import Project=`*.props` />
+            </Project>",
+        };
+
+        [Theory]
+        [MemberData(nameof(ContextPinsGlobExpansionCacheData))]
+        public void ContextCachesImportGlopExpansions(EvaluationContext.SharingPolicy policy, string[][] expectedGlobExpansions)
```
Comment (rainersigwald) [inline src/Build.UnitTests/Definition/ProjectEvaluationContext_Tests.cs]: here too

File: src/Build/Evaluation/Context/EvaluationContext.cs [hunk 3]
```diff
@@ -32,17 +33,23 @@ public enum SharingPolicy
...
-        internal IFileSystem FileSystem { get; } = new CachingFileSystemWrapper(FileSystems.Default);
+        internal ISdkResolverService SdkResolverService { get; }
+        internal IFileSystem FileSystem { get; }
+        internal EngineFileUtilities EngineFileUtilities { get; }
 
         /// <summary>
         /// Key to file entry list. Example usages: cache glob expansion and intermediary directory expansions during glob expansion.
         /// </summary>
-        internal ConcurrentDictionary<string, ImmutableArray<string>> FileEntryExpansionCache = new ConcurrentDictionary<string, ImmutableArray<string>>();
+        internal ConcurrentDictionary<string, ImmutableArray<string>> FileEntryExpansionCache { get; }
```
Comment (rainersigwald) [inline src/Build/Evaluation/Context/EvaluationContext.cs]: Why move these into the ctor? Isn't the effect the same?

Comment (cdmihai) [reply src/Build/Evaluation/Context/EvaluationContext.cs]: I didn't like having part of the initialization code in static blocks and another part in the constructor, so I moved everything in the constructor.

**Pull request merged**